timeseries/src/autogluon/timeseries/trainer/abstract_trainer.py

```diff
@@ -936,6 +940,88 @@ def evaluate(
+        time_start = time.time()
+        if subsample_size > data.num_items:
+            logger.info(
+                f"Subsample_size {subsample_size} is larger than the number of items in the data and will be ignored"
+            )
+            subsample_size = data.num_items
+
+        # set default number of iterations andcap iterations if the number of items in the data is smaller
+        # than the subsample size for the naive method
+        num_iterations = num_iterations or (5 if method == "permutation" else 1)
+        if method == "naive" and data.num_items <= subsample_size:
+            num_iterations = 1
+
+        # if features are provided, check that they are valid features in the covariate metadata
+        if features is None:
+            features = self.metadata.all_features
+        else:
+            if len(features) == 0:
+                raise ValueError(
+                    "No features provided to compute feature importance. At least some valid features should be provided."
+                )
+            for fn in features:
+                if fn not in self.feature_generator.covariate_metadata.all_features:
+                    raise ValueError(f"Feature {fn} not found in covariate metadata.")
+
+        # initialize the importance transform
+        importance_transform_type = {
+            "permutation": PermutationFeatureImportanceTransform,
+            "naive": ConstantReplacementFeatureImportanceTransform,
+        }.get(method)
+        importance_transform = importance_transform_type(covariate_metadata=self.metadata)
+
+        # persist trainer to speed up repeated inference
+        persisted_models = self.persist(model_names=[model] if model else "best", with_ancestors=True)
+
+        relative_drops = defaultdict(list)
+        for n in range(num_iterations):
+            if subsample_size < data.num_items:
+                item_ids_sampled = data.item_ids.to_series().sample(subsample_size)  # noqa
+                data_sample = data.query("item_id in @item_ids_sampled")
+            else:
+                data_sample = data
+
+            base_score = self.evaluate(data=data_sample, model=model, metrics=metric).get(metric.name)
+
+            for feature in features:
+                data_sample_replaced = importance_transform.transform(data_sample, feature_name=feature)
+                score = self.evaluate(data=data_sample_replaced, model=model, metrics=metric).get(metric.name)
+                relative_drops[feature].append((score - base_score) / -base_score)
+
+            if time_limit is not None and time.time() - time_start > time_limit:
+                logger.info(f"Time limit reached, stopping feature importance computation after {n} iterations")
+                break
+
+        self.unpersist(model_names=persisted_models)
+
+        return (
+            pd.DataFrame(relative_drops).agg(["mean", "std"]).T.rename(columns={"mean": "importance", "std": "stdev"})
+        )
```

Comment:
What happens if a static feature & covariate have the same name? Should we add some suffix/prefix to the static features in that case?

Reply:
Thanks for this callout! Would this clash be handled by the feature generator otherwise?

Reply:
We currently don't handle this in the inner loop of feature importance either.

Reply:
wontfix for now, we will be raising a warning